### DISCUSSION TITLE
Using X-init to initialize Google autocomplete on looped dom elements

### DISCUSSION THREAD

**Discussion by jrousan** [livewire/livewire#5412] [1. Help]
So I have a form that allows users to add multiple locations for businesses.  For the address field I want to use Google autocomplete (using NPM package: https://www.npmjs.com/package/google-address-autocomplete). 

In a JS file I have

```
import AddressAutocomplete  from 'google-address-autocomplete';

export default (index) => {
    return {
        init() {
                new AddressAutocomplete('.google-address-lookup', (results) => {
                    this.$wire.set('providerLocations.' + index + '.full_address', results.formattedAddress);
                });
        }
    }
}

```

And in the blade file

                    @foreach($providerLocations as $index => $location)
                    //...
                                <div
                                    x-data="googleSearch('{{$index}}')">
                                    <x-input
                                        wire:model="providerLocations.{{$index}}.full_address" icon="location-marker"
                                        label="Full Address" placeholder="Full Address" class="google-address-lookup" type="search" autocomplete="off"/>
                                </div>
                    //...
                    @endforeach

This works perfectly for the first input box.  It does not work for subsequent search boxes that are loaded via livewire when a user clicks "add location". 

Any suggestions?

**Comment by joshhanley** [comment]
@jrousan I've never used it before, but I suspect it is because you are referencing `.google-address-lookup` which isn't unique. I would try adding an `id` attribute to the input that is unique for each instance and referencing that in the constructor of the autocomplete.

Hope this helps!

**Reply by jrousan** [reply]
Thanks for the reply.  That was my initial instinct and I did append the {{ $index }} to the class name just to see but to no avail. 

I can only suspect the init function itself, maybe I'm misusing it by trying to init on new DOM elements after alpine has already loaded?  

Racking my brain for a work around.

**Reply by t1m0t** [reply]
@joshhanley suggested to add an `id` attribute like `id="someUniqueValue"` in the DOM.

**Reply by joshhanley** [reply]
@jrousan yeah I would suggest testing it using the ID attribute instead and ensuring that they are unique to see if that works.

**Reply by jrousan** [reply]
@joshhanley Yeah, I tried to override the livewire ID like this:

```
    <div
          x-data="googleSearch('{{$index}}')"
          class="col-span-8 lg:col-span-5">
                 <x-input
                  id="id-{{$index}}"
                  wire:model="providerLocations.{{$index}}.full_address" icon="location-marker"
                  label="Full Address" placeholder="Full Address" class="google-address-lookup"/>
    </div>
```

Then passed that in to the JS

```
export default (index) => {
    return {
        init() {
                new AddressAutocomplete('#id-' + index, (results) => {
                    console.log(results.formattedAddress)
                    this.$wire.set('providerLocations.' + index + '.full_address', results.formattedAddress);
                });
        }
    }
}
```

But this yielded the same results.  The autocomplete works on the first field, but when adding a new field, the autocomplete does not initialize.

**Reply by joshhanley** [reply]
@jrousan hmm ok, any chance you could get an example running on https://laravelplayground.com/ and I can have a look? Thanks!
